textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexMatcherTest.java

```diff
@@ -26,46 +26,141 @@
-            TestConstants.SAMPLE_SCHEMA_PEOPLE.get(0).getFieldName());
-    }
-    
-    @After
-    public void cleanUp() throws Exception{
-        dataWriter.clearData();
-    }
-    
-    @Test
-    public void testGetNextTuple() throws Exception{
-        String regex = "b.*"; //matches bruce and brad
-        String fieldName = TestConstants.FIRST_NAME;
-        IPredicate predicate = new RegexPredicate(regex, fieldName);
-        ISourceOperator sourceOperator = new ScanBasedSourceOperator(dataReader);
-        List<ITuple> tuples = TestConstants.getSamplePeopleTuples();
-        
-        regexMatcher = new RegexMatcher(predicate, sourceOperator);
-        regexMatcher.open();
-        ITuple nextTuple = null;
-        int numTuples = 0;
-        while((nextTuple = regexMatcher.getNextTuple()) != null){
-            boolean contains = TestUtils.contains(tuples, nextTuple);
-            Assert.assertTrue(contains);
-            numTuples ++;
-        }
-        Assert.assertEquals(2, numTuples);
-        regexMatcher.close();
-    }
-
-}
\ No newline at end of file
+
+	private RegexMatcher regexMatcher;
+	private IDataWriter dataWriter;
+	private IDataReader dataReader;
+	private IDataStore dataStore;
+
+	public void setUpPeople() throws Exception {
+		dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR,
+				TestConstants.SAMPLE_SCHEMA_PEOPLE);
+		dataWriter = new LuceneDataWriter(dataStore);
+		dataWriter.clearData();
+		dataWriter.writeData(TestConstants.getSamplePeopleTuples());
+	}
+
+	public void setUpCorp() throws Exception {
+		dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR,
+				TestConstants.SAMPLE_SCHEMA_CORP);
+		dataWriter = new LuceneDataWriter(dataStore);
+		dataWriter.clearData();
+		dataWriter.writeData(TestConstants.getSampleCorpTuples());
+	}
+
+	public void setUpStaff() throws Exception {
+		dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR,
+				TestConstants.SAMPLE_SCHEMA_STAFF);
+		dataWriter = new LuceneDataWriter(dataStore);
+		dataWriter.clearData();
+		dataWriter.writeData(TestConstants.getSampleStaffTuples());
+
+	}
+
+	@After
+	public void cleanUp() throws Exception {
+		dataWriter.clearData();
+	}
+
+	
+	@Test
+	public void testNameGetNextTuple() throws Exception {
+		setUpPeople();
+		dataReader = new LuceneDataReader(dataStore,
+				LuceneConstants.SCAN_QUERY, TestConstants.FIRST_NAME);
+
+		String regex = "b.*"; // matches bruce and brad
+		String fieldName = TestConstants.FIRST_NAME;
+		IPredicate predicate = new RegexPredicate(regex, fieldName);
+		ISourceOperator sourceOperator = new ScanBasedSourceOperator(dataReader);
+		List<ITuple> tuples = TestConstants.getSamplePeopleTuples();
+
+		regexMatcher = new RegexMatcher(predicate, sourceOperator);
+		regexMatcher.open();
+		ITuple nextTuple = null;
+		int numTuples = 0;
+		while ((nextTuple = regexMatcher.getNextTuple()) != null) {
+			boolean contains = TestUtils.contains(tuples, nextTuple, TestConstants.SAMPLE_SCHEMA_PEOPLE);
```

Comment:
Checking the number of expected results is NOT enough.  The test case should include exactly the tuples that are expected, not just a number.

A general suggestion: it will be very useful to have a helper function that includes a set of tuples, a schema, a regex query, and expected results.  Then you can easily produce many test cases by providing the right parameters. 

It's very important to have many, and good test cases.